pysages/backends/hoomd.py

```diff
@@ -16,27 +16,26 @@
 from pysages.backends.common import HelperMethods
 from pysages.backends.snapshot import Box, Snapshot
 
+from .core import ContextWrapper, Sampler
 
-class ContextWrapper:
+
+class ContextWrapperHOOMD(ContextWrapper):
+    """
+    ContextWrapper for hoomd-blue.
+    """
     def __init__(self, context):
+        super().__init__(context)
         self.sysview = SystemView(context.system_definition)
-        self.context = context
         self.synchronize = self.sysview.synchronize
 
 
-class Sampler(HalfStepHook):
-    def __init__(self, method_bundle, bias):
-        super().__init__()
-        #
-        snapshot, initialize, update = method_bundle
-        self.snapshot = snapshot
-        self.state = initialize()
-        self._update = update
-        self.bias = bias
-    #
-    def update(self, timestep):
-        self.state = self._update(self.snapshot, self.state)
-        self.bias(self.snapshot, self.state)
+class SamplerHOOMD(Sampler, HalfStepHook):
+    """
+    Sampler class for hoomd-blue backend.
+    """
+    def __init__(self, method_bundle, bias, callback=None):
+        HalfStepHook.__init__()
+        Sampler.__init__(method_bundle, bias, callback)
```

Comment:
For HOOMD, `update` needs to be defined and take a `timestep` argument (we are mirroring a C++ class here). I could modify the OpenMM C++ plugin to fit this interface to make them uniform. But for now we need to special case either one or the other.

Reply:
Yes, that is an issue, that hoomd and openmm differ here.
But I still don't think the callback needs an explicit timestep.
Doesn't snapshot have an attribute for timestep?
And if it doesn't, we should add one.

Reply:
I am strongly voting to unify this somehow.

Reply:
I will look into it, but that has to be done on the `openmm-dlext` side. I opened https://github.com/SSAGESLabs/openmm-dlext/issues/6 to remind us.